src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -1,332 +1,123 @@
         }
 
-        private bool TryNegateBinaryComparisonExpression(
-            ExpressionSyntax expression,
-            SemanticModel semanticModel,
-            CancellationToken cancellationToken,
-            out ExpressionSyntax result)
+        private void GetPartsOfIfStatement(SyntaxNode node, out SyntaxNode condition, out SyntaxNode thenStatements, out SyntaxNode elseStatements)
         {
-            if (s_binaryMap.TryGetValue(expression.Kind(), out var tuple))
-            {
-                var binaryExpression = (BinaryExpressionSyntax)expression;
-                var expressionType = tuple.Item1;
-                var operatorType = tuple.Item2;
-
-                // Special case negating Length > 0 to Length == 0 and 0 < Length to 0 == Length
-                // for arrays and strings. We can do this because we know that Length cannot be
-                // less than 0. Additionally, if we find Length == 0 or 0 == Length, we'll invert
-                // it to Length > 0 or 0 < Length, respectively.
-                if (IsComparisonOfZeroAndSomethingNeverLessThanZero(binaryExpression, semanticModel, cancellationToken))
-                {
-                    operatorType = binaryExpression.OperatorToken.Kind() == SyntaxKind.EqualsEqualsToken
-                        ? binaryExpression.Right is LiteralExpressionSyntax ? SyntaxKind.GreaterThanToken : SyntaxKind.LessThanToken
-                        : SyntaxKind.EqualsEqualsToken;
-                    expressionType = binaryExpression.Kind() == SyntaxKind.EqualsExpression
-                        ? binaryExpression.Right is LiteralExpressionSyntax ? SyntaxKind.GreaterThanExpression : SyntaxKind.LessThanExpression
-                        : SyntaxKind.EqualsExpression;
-                }
+            var ifStatement = (IfStatementSyntax)node;
+            condition = ifStatement.Condition;
+            thenStatements = ifStatement.Statement;
+            elseStatements = ifStatement.Else;
+        }
 
-                result = SyntaxFactory.BinaryExpression(
-                    expressionType,
-                    binaryExpression.Left,
-                    SyntaxFactory.Token(
-                        binaryExpression.OperatorToken.LeadingTrivia,
-                        operatorType,
-                        binaryExpression.OperatorToken.TrailingTrivia),
-                    binaryExpression.Right);
+        internal override ISyntaxFactsService GetSyntaxFactsService()
+           => CSharpSyntaxFactsService.Instance;
 
-                return true;
-            }
-
-            result = null;
-            return false;
+        internal override string GetInvertIfText()
+        {
+            return CSharpFeaturesResources.Invert_if;
         }
 
-        private ExpressionSyntax Negate(ExpressionSyntax expression, SemanticModel semanticModel, CancellationToken cancellationToken)
+        internal override bool IsConditionalAnd(SyntaxToken operatorToken)
```

Comment:
do you need to check the token?  wouldn't you have a binary expression to begin with that you could check?